app/models/ability.rb

```diff
@@ -1,80 +1,13 @@
 # frozen_string_literal: true
 
-# :nodoc:
+# Defines the Ability class that delegates to UserAbility or TokenAbility
+# based on presence of a token
 class Ability
-  include CanCan::Ability
-
-  attr_reader :allowlisted_jwt, :user
-
-  # rubocop:disable Metrics/CyclomaticComplexity
-  # rubocop:disable Metrics/PerceivedComplexity
-  def initialize(user, token = nil)
-    alias_action :create, :read, :update, :destroy, to: :crud
-
-    can :confirm, ContactEmail
-    return unless user || token
-
-    @user = user
-
-    if token
-      can :read, Organization, allowlisted_jwts: { jti: token['jti'] }
-
-      @allowlisted_jwt = AllowlistedJwt.find_by(jti: token['jti'])
-
-      case @allowlisted_jwt.scope
-      when 'all'
-        can %i[create update], [Stream, Upload], organization: { allowlisted_jwts: { jti: token['jti'] } }
-        can :read, Organization, public: true
-        can :read, [Stream, Upload], organization: { public: true }
-
-        can :read, ActiveStorage::Attachment, { record: { organization: { public: true } } }
-      when 'upload'
-        can %i[create update], [Stream, Upload], organization: { allowlisted_jwts: { jti: token['jti'] } }
-      when 'download'
-        can :read, Organization, public: true
-        can :read, [Stream, Upload], organization: { public: true }
-
-        can :read, ActiveStorage::Attachment, { record: { organization: { public: true } } }
-      end
-
-      allowlisted_jwt&.update(updated_at: Time.zone.now)
-      return
-    end
-
-    if user.roles.any?
-      can :read, ActiveStorage::Attachment, { record: { organization: { public: true } } }
-      can :read, MarcRecord, upload: { organization: { public: true } }
-      can %i[read profile normalized_data processing_status], Stream, organization: { public: true }
-      can %i[read info], Upload, organization: { public: true }
-      can :read, :pages_data
-      can %i[read users organization_details provider_details], Organization, public: true
+  def self.new(user, token = nil)
+    if token.present?
+      Abilities::TokenAbility.new(token)
+    else
+      Abilities::UserAbility.new(user)
```

Comment:
I haven't thought about this PR much, but overriding `new` feels a little funny. Maybe this condition is better in https://github.com/pod4lib/aggregator/blob/main/app/controllers/concerns/custom_pod_ability_concern.rb#L7?